pages/search.js

```diff
@@ -95,6 +93,14 @@ export default function Search({ data: { tags, randUsers }, BASE_URL }) {
 
     const timer = setTimeout(() => {
       fetchUsers(inputValue);
+      const newUrl = `/search?keyword=${inputValue}`;
+
+      // replace the url with the new url without reloading the page
+      window.history.replaceState(
```

Comment:
I think react has builtin functionality for this

Reply:
Thank you for your response. I wasn't aware of any built-in functionality in React for updating the URL without reloading the page. I tried using Next.js' router.replace, but it reloads the page, causing redundant API calls and useEffect re-renders. Could you please provide more information or guidance on the React built-in functionality you mentioned? Any specific resources would be greatly appreciated.

Reply:
@eddiejaoude can we please have a update on this

Reply:
router push will reload the page and also add a new entry to the browser history stack. if we don't need to bother about extra reload of the page, router `replace` would be the best choice as it won't add a new entry into the history stack. please let me know what I can follow here.

Reply:
Yes I think it is better to go onto the history stack.

> Shallow routing allows you to change the URL without running data fetching methods again. The URL will get updated to /?counter=10 and the page won't get replaced, only the state of the route is changed.

You can use `push` without a page reload using `shallow` option, you can read the official docs here https://nextjs.org/docs/pages/building-your-application/routing/linking-and-navigating#shallow-routing

Reply:
already tried using push with shallow option

Reply:
We could move it out of the `use effect` into a function that is triggered by the input field directly?